scripts/tests/test_phase43_source_root_feasibility_figure.py

```
@@ -0,0 +1,78 @@
+                "verified": "true",
+            }
+            row.update(overrides.get(variant, {}))
+            rows.append(row)
+        return rows
+
+    def test_validate_rows_accepts_measured_median_metadata(self):
+        steps = MODULE.validate_rows(self.sample_rows(), source=Path("sample.tsv"))
+        self.assertEqual(steps, [2])
+
+    def test_validate_rows_rejects_even_median_timing_runs(self):
+        with self.assertRaises(SystemExit) as ctx:
```

Ruff (0.15.11)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)